### DISCUSSION TITLE
Design Discussion

### DISCUSSION THREAD

**Discussion by hasnainmakada-99** [reactplay/react-play#81] [Q&A] [answered]
hello @nirmalkc , I want to discuss regarding the design which we will implement in [43](https://github.com/atapas/react-play/issues/43)

**Comment by atapas** [comment, marked as answer]
@hasnainmakada-99 I suggest you develop the component that shows the message of "No Play Found" and complete the logic of showing it conditionally. Once the functionality is done, @nirmalkc will style that component. Just notify here. Thanks!

**Reply by hasnainmakada-99** [reply]
ok sir, ill start the designing

**Reply by hasnainmakada-99** [reply]
Tapas sir i've found the file in which the search functionality is there, and i've tried to out conditions around it, but it is not working, but I'll try to resolve it

**Reply by atapas** [reply]
Sure thanks, btw I suggested you complete the code implementation and let @nirmalkc do the required design and style of the component's look and feel. Just clarifying to be on the same page :)

**Reply by hasnainmakada-99** [reply]
yes sir I m trying to implement the logic only

**Reply by hasnainmakada-99** [reply]
and sir only one file named as serch-play.jsx implements the search functionality no ??

**Reply by hasnainmakada-99** [reply]
![image](https://user-images.githubusercontent.com/82728823/164387569-39cc1720-b017-4dd5-81f7-4f2d3e2b0828.png)


tapas sir for testing i've copied the image of page not found and i've made some changes & and the fucntionality is working properly

**Reply by hasnainmakada-99** [reply]
when we hit enter in the search if the plays does not match it will throw this error
now the remaining part is the design, which is to be implemented

**Reply by hasnainmakada-99** [reply]
@atapas  sir , sir i have a doubt how to i get the search filter the previous functionality was working on location changing , now I am not able to filter the search results

**Reply by atapas** [reply]
Put your changes in PR so that I can see code changes. Let's discuss on PR.

**Reply by hasnainmakada-99** [reply]
Ok sir

**Reply by atapas** [reply]
@hasnainmakada-99 The fix should be only in one file `PlayList.jsx`

Please check for the `plays` array to be null/undefined/empty and call your component to show the message conditionally.

```js
<>
      <ol className="list-plays">
        {plays.map((play, index) => (
          <PlayThumbnail key={play.id} play={play} />
        ))}
      </ol>
    </>
```

Please raise the PR.

**Reply by hasnainmakada-99** [reply]
Ok sir ill raise the pr

**Reply by hasnainmakada-99** [reply]
tapas sir, I've raised the pr please review it, and the functionality is still not working i've tested it with various conditions

**Comment by hasnainmakada-99** [comment]
@atapas sir now i will have to work only on the playlist.jsx file no ??

**Reply by atapas** [reply]
Yes tht's it.

**Reply by hasnainmakada-99** [reply]
sir i've checked the plays objects with various conditions but it is not working at all

**Reply by hasnainmakada-99** [reply]
i am trying to print a simple message also that play not found, but it is not working at all

**Reply by hasnainmakada-99** [reply]
tapas sir its working

**Reply by hasnainmakada-99** [reply]
yaaaaaayyy

**Reply by hasnainmakada-99** [reply]
![image](https://user-images.githubusercontent.com/82728823/164624659-72ba7afa-e864-4d2e-8805-0fcc7a7e1529.png)

**Reply by hasnainmakada-99** [reply]
see @atapas sir

**Reply by hasnainmakada-99** [reply]
sir i've checked all the conditions and its working perfectly

**Reply by hasnainmakada-99** [reply]
![image](https://user-images.githubusercontent.com/82728823/164625180-e0785440-9ce6-4c37-a92a-0d6b028f304c.png)

**Reply by hasnainmakada-99** [reply]
here is the working functionality

**Reply by atapas** [reply]
Cool, open a PR

**Reply by hasnainmakada-99** [reply]
sir the desgin will noew be reamaining

**Reply by hasnainmakada-99** [reply]
@atapas sir please review my pr

**Comment by ali0sam** [comment]
hi, it's not nice to we have a dark background color for ``/plays`` ? or just a theme changer button?

![Screenshot from 2022-04-24 01-32-27](https://user-images.githubusercontent.com/69610848/164945895-3f44f7f6-ca76-4344-8418-76e3b7161f21.png)

**Reply by atapas** [reply]
@ali0sam I think it should be take as part of theming. Thanks for the callout. I'll create an issue for theme support.
